main.py

```diff
@@ -1,37 +1,87 @@
+from ast import Pass
 from flask import Flask, redirect, url_for, render_template
+from flask_bootstrap import Bootstrap
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask_login import LoginManager, UserMixin, login_user, login_required, logout_user, current_user
+from flask import flash
+from Forms import LoginForm, SignUpForm
 
 app = Flask(__name__)
+app.config['SECRET_KEY'] = 'imageanalysissystem'
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///database.db'
+Bootstrap(app)
 
+db = SQLAlchemy(app)
+
+loginManager = LoginManager()
+loginManager.init_app(app)
+loginManager.login_view = 'login'
+
+class User(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(50))
+    email = db.Column(db.String(50), unique=True)
+    password = db.Column(db.String(80))
+    
+@loginManager.user_loader
+def loadUser(id):
+    return User.query.get(int(id))
+
+@app.route("/home")
 @app.route("/")
+@login_required
 def home():
-    return render_template("home.html")
+    return render_template("home.html", name=current_user.name)
+
+@app.route("/login", methods=['GET', 'POST'])
+def login():
+    form = LoginForm()
+
+    if form.validate_on_submit():
+        user = User.query.filter_by(email=form.email.data).first()
+        if user:
+            if check_password_hash(user.password, form.password.data):
+                login_user(user)
+                return redirect(url_for('home'))
+        flash("Invalid email/password")
+    return render_template("login.html", form=form)
+
+@app.route("/signup", methods=['GET', 'POST'])
+def signup():
+    form = SignUpForm()
+
+    if form.validate_on_submit():
+        hashedPassword = generate_password_hash(form.password.data, method='sha256')
+        newUser = User(name=form.name.data, email= form.email.data, password=hashedPassword)
+
+        db.session.add(newUser)
+        db.session.commit()
+
+        return redirect(url_for('login'))
+
+    return render_template("signup.html", form=form)
+
+@app.route('/logout')
+@login_required
+def logout():
+    logout_user()
+    return redirect(url_for('login'))
 
 @app.route("/events")
+@login_required
```

Comment:
This `@login_required` tag ensures that this page is only able to be accessed from a **logged in user**. You can't just type in this route in the URL to access this page without logging in first.